bindings/matrix-sdk-ffi/src/timeline/mod.rs

```diff
@@ -710,6 +714,24 @@ impl Timeline {
     }
 }
 
+/// Given a pair of optional `caption` and `formatted_caption` parameters,
+/// return a formatted caption:
+///
+/// - If a `formatted_caption` exists, return it.
+/// - If it doesn't exist but there is a `caption`, parse it as markdown and
+///   return the result.
+/// - Return `None` if there are no `caption` or `formatted_caption` parameters.
+fn formatted_caption_from(
+    caption: &Option<String>,
+    formatted_caption: &Option<FormattedBody>,
+) -> Option<RumaFormattedBody> {
+    match (&caption, formatted_caption) {
+        (None, None) => None,
+        (Some(body), None) => RumaFormattedBody::markdown(body),
+        (_, Some(formatted_body)) => Some(formatted_body.clone().into()),
```

Comment:
> I would invert those two branches, otherwise a body and a formatted_body will result in the body overriding the formatted body 🤪

I don't mind, but can't the branch that overrides only happen if there is no formatted body `(body -> Some, formatted_body -> None)`? Maybe I misunderstood how this pattern matching of tuples work.

> …This needs a test 😁

I guess I can move this into the SDK crate then.

Reply:
Ah you're right, sorry I misread that. That would then be clearer as a series of if let some return, maybe?

Reply:
> Doesn't matter much but I find the match much clearer

I just changed it 😭 .

Reply:
Ah sorry, though I don't really think you need to change it back. I'm just saying that using a match like that is completely fine and even preferred by me.

Reply:
I also found it more expressive. Since it states all 3 possible causes and the states of the variables for those. Anyway, it's not a big deal as you said.

Reply:
I was definitely confused and misread it, I do find the new code much much clearer since it's in-your-face at the beginning that the formatted body is preferred over anything else, so YMMV, heh :)